src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -874,6 +874,13 @@ private static void ClearAndReturn(ArraySegment<byte> rented)
             }
         }
 
+        internal void WritePropertyName(int index, Utf8JsonWriter writer)
+        {
+            CheckNotDisposed();
```

Comment:
Given that this is marked internal, would it be possible to add some kind of assertion checking that the index does correspond to a property name?